src/components/PopupWithForm.js

```diff
@@ -17,15 +18,38 @@ export default class PopupWithForm extends Popup {
   }
 
   reset() {
-    this._popupForm.reset();
+    if (this._popupForm) {
+      this._popupForm.reset();
+    }
   }
 
   setEventListeners() {
-    this._popupForm.addEventListener("submit", (e) => {
-      e.preventDefault();
-      this._handleFormSubmit(this._getInputValues());
-    });
-
     super.setEventListeners();
+    if (this._popupForm) {
+      this._popupForm.addEventListener("submit", (e) => {
+        e.preventDefault();
+        this._handleFormSubmit(this._getInputValues());
+      });
+    }
+  }
+
+  open() {
+    super.open();
+  }
+
```

Comment:
**Could be improved**

You can make ` setInputValues` in class `PopupWithForm`, which can insert data into inputs:

```js

  setInputValues(data) {
    this._inputList.forEach((input) => {
      // here you insert the `value` by the `name` of the input
      input.value = data[input.name];
    });
  }



```

So, this way you’ll not have to search the inputs of the profile in `index.js`